Tools/Algorithms/stencil.py

```diff
@@ -407,51 +406,37 @@ def run_main(dx, dy, dz, dt, nox, noy, noz, gamma=1., galilean=False,
-    gc_nodal_x = compute_guard_cells(ex, stencils['x']['nodal'])
-    gc_nodal_y = compute_guard_cells(ey, stencils['y']['nodal'])
-    gc_nodal_z = compute_guard_cells(ez, stencils['z']['nodal'])
-
-    print('\n    nodal solver:')
-    print('    - {:d} guard cells along x'.format(gc_nodal_x))
-    print('    - {:d} guard cells along y'.format(gc_nodal_y))
-    print('    - {:d} guard cells along z'.format(gc_nodal_z))
-
-    # 2) Staggered solver
-    gc_stagg_x = compute_guard_cells(ex, stencils['x']['stagg'])
-    gc_stagg_y = compute_guard_cells(ey, stencils['y']['stagg'])
-    gc_stagg_z = compute_guard_cells(ez, stencils['z']['stagg'])
-
-    print('\n    staggered solver:')
-    print('    - {:d} guard cells along x'.format(gc_stagg_x))
-    print('    - {:d} guard cells along y'.format(gc_stagg_y))
-    print('    - {:d} guard cells along z'.format(gc_stagg_z))
-
     # Plot stencils
-    plot_stencil(cx, stencils['x']['nodal'], stencils['x']['stagg'],
-                 'x', path, name)
-    plot_stencil(cy, stencils['y']['nodal'], stencils['y']['stagg'],
-                 'y', path, name)
-    plot_stencil(cz, stencils['z']['nodal'], stencils['z']['stagg'],
-                 'z', path, name)
-
-    print('\n >> Output Summary')
-    print('    --------------')
-
-    print('\n    path to figures: \'' + os.path.abspath(path) + '\'')
-
-    print('\n    stencil arrays:')
-    print('    - sx_nodal')
-    print('    - sx_stagg')
-    print('    - sy_nodal')
-    print('    - sy_stagg')
-    print('    - sz_nodal')
-    print('    - sz_stagg\n')
+    plot_stencil(cx, stencils['x']['nodal'], stencils['x']['stagg'], 'x', path, name)
+    plot_stencil(cy, stencils['y']['nodal'], stencils['y']['stagg'], 'y', path, name)
+    plot_stencil(cz, stencils['z']['nodal'], stencils['z']['stagg'], 'z', path, name)
+
+    # Compute min and max numbers of guard cells
+    gcmin_x_nodal, gcmax_x_nodal = compute_guard_cells(sp, dp, stencils['x']['nodal'])
+    gcmin_y_nodal, gcmax_y_nodal = compute_guard_cells(sp, dp, stencils['y']['nodal'])
+    gcmin_z_nodal, gcmax_z_nodal = compute_guard_cells(sp, dp, stencils['z']['nodal'])
+    #
+    gcmin_x_stagg, gcmax_x_stagg = compute_guard_cells(sp, dp, stencils['x']['stagg'])
+    gcmin_y_stagg, gcmax_y_stagg = compute_guard_cells(sp, dp, stencils['y']['stagg'])
+    gcmin_z_stagg, gcmax_z_stagg = compute_guard_cells(sp, dp, stencils['z']['stagg'])
+
+    fig_path = os.path.abspath(path)
+    print(f'\nFigures saved in {fig_path}/.')
+    print('\nThe plots show the extent of the signal to be truncated (y-axis)'
+        + '\nby choosing a given number of cells (x-axis) for the ghost regions'
+        + '\nof each simulation grid, along x, y, and z.')
+    print('\nIt is recommended to choose a number of ghost cells that corresponds to'
+        + '\na truncation of the signal between single and double machine precision.'
+        + '\nThe more guard cells, the more accurate, yet expensive, results.')
```

Comment:
It might be helpful to state what the shaded region is even though it should be clear from context.
```suggestion
        + '\nThe more guard cells, the more accurate, yet expensive, results.'
        + '\nFor each stencil the region of accuracy between single and double'
        + '\nprecision is shaded to help you identify a suitable number of'
        + '\nghost cells.`)
```